compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -231,7 +231,9 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
 
         // We now fake capture information for all variables that are mentioned within the closure
```

Comment:
Can you update this comment to include an explanation for the added logic? And also add a reference to the issue number?